### PR TITLE
sync the response message unconditionally

### PR DESCRIPTION
**Pull request by sdghchj**
if the operation is response, force to sync the resource message, since the edgecore requests it

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
In case that the edgecore with the same node name is re-installed or moved to another edge node, it can't find imagePullSecrets or configMaps in local metamanager and also failed to get response from cloudcore.
If edgecore request a resouce from cloudcore, which means the resource doesn't exist in local metamanager, cloudcore has to sync the response resource down to edge.

**Which issue(s) this PR fixes**:
#2967

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

**Review by fisherxu — COMMENTED** [review]
@sdghchj Thanks for the fixing, and could you please rebase to the master branch to fix the failed e2e ci?

**Comment by sdghchj** [comment]
> @sdghchj Thanks for the fixing, and could you please rebase to the master branch to fix the failed e2e ci?

@fisherxu   Rebased.

File: cloud/pkg/cloudhub/channelq/channelq.go [hunk 1]
```diff
@@ -100,9 +100,10 @@ func (q *ChannelMessageQueue) addMessageToQueue(nodeID string, msg *beehiveModel
 		return
 	}
 
-	item, exist, _ := nodeStore.GetByKey(messageKey)
-
-	if !isDeleteMessage(msg) {
+	//if the operation is delete, force to sync the resource message
+	//if the operation is response, force to sync the resource message, since the edgecore requests it
+	if !isDeleteMessage(msg) && msg.GetOperation() != beehiveModel.ResponseOperation {
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/channelq/channelq.go:105]: Thanks  for the fixing!

Here the `if !isDeleteMessage(msg)` means if the message is not delete message, we will do some check like remove the dumplicate message. And if the message is delete message, we will send it directly.

And the Response Message we will also send to edge at here: https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/cloudhub/channelq/channelq.go#L168

So this is not the root cause of the issue.

Comment (sdghchj) [reply cloud/pkg/cloudhub/channelq/channelq.go:105]: Yes, if `exist` is true which means the duplicate message in nodeStore, as you mentioned,  some checks will be done like remove the dumplicate message. But if `exist` is false, I see `objectSync` may prevent the message to be send to edge: https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/cloudhub/channelq/channelq.go#L116.

And as a reponse to query a secret,  the field `content` stores the replied secret object, not string `OK`, so the boolean value of the experession at https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/cloudhub/channelq/channelq.go#L170 will be false which means the message is not list resource.

At last, I have tested my PR,  and it did work.

Comment (fisherxu) [reply cloud/pkg/cloudhub/channelq/channelq.go:105]: Yes, I just double check the issue, it's really bug in this scenario, we can have more discussions.

Comment (sdghchj) [reply cloud/pkg/cloudhub/channelq/channelq.go:105]: What to do next ?  @fisherxu

Comment (fisherxu) [reply cloud/pkg/cloudhub/channelq/channelq.go:105]: If the edgenode is crashed or reinstalled(data in db lost), the pod will also can't recovered, right? I think we need a design or full solution for it?

**Comment by sdghchj** [comment]
> If the edgenode is crashed or reinstalled(data in db lost), the pod will also can't recovered, right? I think we need a design or full solution for it?
@fisherxu  
Actually, If one uninstalled edgecore, he had to be aware of  the effect made by his behavior.
To me, I can accept the fact that I have to re-deploy all the pods manually, but I am disappointed to see the  pods failed to run because of related resources such as secrets、configMaps.

Of course, I have a suggestion that edgecore also query the pods from cloud when starting if no pods found in local db ,and cloud reponse all the pods down.

**Comment by fisherxu** [comment]
> Of course, I have a suggestion that edgecore also query the pods from cloud when starting if no pods found in local db ,and cloud reponse all the pods down.

This patch can fix restarting a new node with same name, and send cm\secret to edge, we can get it merged. :) And for edge recovering from edgenode crashed, we can have further design.

**Comment by sdghchj** [comment]
> > Of course, I have a suggestion that edgecore also query the pods from cloud when starting if no pods found in local db ,and cloud reponse all the pods down.
> 
> This patch can fix restarting a new node with same name, and send cm\secret to edge, we can get it merged. :) And for edge recovering from edgenode crashed, we can have further design.

OK

**Review by fisherxu — COMMENTED** [review]
/approve

**Pull request merged**